### PR TITLE
Cleanup MapEvaluator

### PR DESCRIPTION
**Pull request by QRemy**
Cleanup MapEvaluator

### CODE REVIEW

**Review by bkhelifi — APPROVED** [review]
Thanks a lot @QRemy

File: gammapy/datasets/evaluator.py [hunk 1]
```diff
@@ -318,14 +300,10 @@ def _compute_flux_spatial(self):
...
 
-            wcs_geom = self.geom.to_wcs_geom(width_min=self.cutout_width).to_image()
+            wcs_geom = self.geom.to_wcs_geom(width_min=self.cutout_width)
+            values = self._compute_flux_spatial_geom(wcs_geom)
 
-            if self.psf and self.model.apply_irf["psf"]:
-                values = self._compute_flux_spatial_geom(wcs_geom)
-            else:
-                values = self.model.spatial_model.integrate_geom(
-                    wcs_geom, oversampling_factor=1
```
Comment (registerrier) [inline gammapy/datasets/evaluator.py:327]: I understand why you remove here because it is basically unused. But `oversampling=1` is a bug right?

**Review by registerrier — COMMENTED** [review]
Thanks @QRemy. I am not sure I completely follow the motivation. There is a bug here right? the oversampling_factor shouldn't be set to 1 when calling integrate_geom.

File: gammapy/datasets/evaluator.py (hunk 1 above)
Comment (QRemy) [reply gammapy/datasets/evaluator.py:327]: yes it is probably a bug, it was not tested before, should I open a separate for that or we backport this PR ?

Comment (registerrier) [reply gammapy/datasets/evaluator.py:327]: there has been a mix-up when working on https://github.com/gammapy/gammapy/pull/3440

Well I would try to see what happens with the correct oversampling factor.

Comment (QRemy) [reply gammapy/datasets/evaluator.py:327]: This line is triggered only for region geom when self.model.apply_irf["psf"] is False. If psf is None we directly return 1 (line 301). With this modification and the test added it also returns 1 for the case self.model.apply_irf["psf"] is False.

Comment (QRemy) [reply gammapy/datasets/evaluator.py:327]: doen in #5390

File: gammapy/datasets/tests/test_evaluator.py [hunk 2]
```diff
@@ -32,8 +32,8 @@ def evaluator():
     )
 
     spectral_model = ConstantSpectralModel()
-    spatial_model = PointSpatialModel(
-        lon_0=0 * u.deg, lat_0=0 * u.deg, frame="galactic"
+    spatial_model = GaussianSpatialModel(
+        lon_0=0 * u.deg, lat_0=0 * u.deg, sigma=0.001 * u.deg, frame="galactic"
```
Comment (QRemy) [inline gammapy/datasets/tests/test_evaluator.py:36]:
```suggestion
    spatial_model = PointSpatialModel(
        lon_0=0 * u.deg, lat_0=0 * u.deg, frame="galactic"```

File: gammapy/datasets/evaluator.py (hunk 1 above)
Comment (QRemy) [reply gammapy/datasets/evaluator.py]: rebased after #5390 so now this PR contains only cleanup

**Pull request merged**